pkg/buildscript/internal/buildexpression/buildexpression.go

```diff
@@ -177,27 +144,29 @@ func New(data []byte) (*BuildExpression, error) {
 	return expr, nil
 }
 
-// NewEmpty creates a minimal, empty buildexpression.
-func NewEmpty() (*BuildExpression, error) {
+// New creates a minimal, empty buildexpression.
+func New() (*BuildExpression, error) {
 	// At this time, there is no way to ask the Platform for an empty buildexpression, so build one
 	// manually.
-	expr, err := New([]byte(`
-		{
-			"let": {
-				"runtime": {
-					"solve_legacy": {
-						"at_time": "$at_time",
-						"build_flags": [],
-						"camel_flags": [],
-						"platforms": [],
-						"requirements": [],
-						"solver_version": null
-					}
-				},
-				"in": "$runtime"
-			}
-		}
-	`))
+	expr, err := Unmarshal([]byte(`
+{
+	"let": {
+		"sources": {
+				"solve": {
+					"at_time": "$at_time",
+					"platforms": [],
+					"requirements": [],
+					"solver_version": null
+				}
+		},
+		"runtime": {
+				"state_tool_artifacts": {
+						"src": "$sources"
+				}
+		},
+		"in": "$runtime"
+	}
+}`))
```

Comment:
Checked in with Martin and he agreed it made sense to update this so we're not invoking anything legacy.